Makefile

```
@@ -35,9 +37,22 @@ luacheck:
 
 .PHONY: test
 test:
-	luatest -v --shuffle all:${SEED}
+	luatest -v --coverage --shuffle all:${SEED}
+
+$(LUACOV_STATS): test
+
+coverage:
+	sed -i -e 's@'"$$(realpath .)"'/@@' $(LUACOV_STATS)
+	cd $(PROJECT_DIR) && luacov test/*.lua
```

It could fix the problem.
```suggestion
	cd $(PROJECT_DIR) && luacov roles/*.lua
```